crates/proc_macro_srv/src/abis/mod.rs

```diff
@@ -69,22 +69,26 @@ impl Abi {
         symbol_name: String,
         info: RustCInfo,
     ) -> Result<Abi, LoadProcMacroDylibError> {
-        if info.version.0 != 1 {
-            Err(LoadProcMacroDylibError::UnsupportedABI)
-        } else if info.version.1 < 47 {
-            Err(LoadProcMacroDylibError::UnsupportedABI)
-        } else if info.version.1 < 54 {
-            let inner = unsafe { Abi_1_47::from_lib(lib, symbol_name) }?;
-            Ok(Abi::Abi1_47(inner))
-        } else if info.version.1 < 56 {
-            let inner = unsafe { Abi_1_55::from_lib(lib, symbol_name) }?;
```

Comment:
Say the minor version is 55, then `< 54` is false but `< 56` is true, so it ends up at this line, using the 1.55 abi.

Reply:
No, let's say the minor is 54.

Reply:
Right, then it still ends up using 1.55, so the old code is wrong. In any case your code is much clearer.